src/StoragelessSession/Session/Data.php

```diff
@@ -25,6 +25,28 @@ private function __construct(array $data, array $metadata)
         $this->metadata = $metadata;
     }
 
+    public static function fromDecodedTokenData(\stdClass $data)
+    {
+        return self::fromTokenData(self::convertStdClassToUsableStuff($data), []);
+    }
+
+    private static function convertStdClassToUsableStuff(\stdClass $shit)
```

Comment:
A long time ago in a galaxy far far away I was using arrays, but one of the users had a problem when creating & verifying a token with an empty object as claim value (I don't know why they used that).

The lib was serializing using ```{}``` and to verify was generating again with ```[]``` which invalidates signature.

I was doing a stupid thing (generating the payload to verify signature) and that was fixed in v3.x. I think that we can go back to arrays in v4 if its better (but that discussion should probably be in JWT lib LOL). Of course you can send a PR ;-)